sdk/python/feast/transformation/pandas_transformation.py

```diff
@@ -19,36 +19,61 @@
 class PandasTransformation(Transformation):
     def __new__(
         cls,
-        udf: Callable[[Any], Any],
-        udf_string: str,
+        udf: Optional[Callable[[Any], Any]] = None,
```

Comment:
why make this optional?

Reply:
It's not optional in real use, we are still maintaining strict validation (lines 72-75 enforce that they can't actually be None). We have to make it optional to ensures that PandasTransformation objects can be properly reconstructed across Ray workers without constructor failures.